rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java

```diff
@@ -0,0 +1,100 @@
+package com.stnetix.ariaddna.externalcloudapi.cloudinterface;
+
+/**
+ * This is an abstract interface containing general methods for
+ * manipulation with resources at any external cloud storage.
+ * Every certain cloud storage realization must implement this interface.
+ */
+public interface iAbstractCloud {
+    /**
+     * An uploadResource method implementation takes as a parameter a path to a resource
+     * which is stored at user's local HDD/SSD and sends it to an external cloud storage.
+     * Notice, that resource may be a file or directory with files.
+     *
+     * @param path A path to a resource at user's local storage.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+
+    String uploadResource(String path);
```

Comment:
1. Do you mean uploadResource(File path) ?
2. I thought that the String as the return type may contain external cloud server responce. I've looked through Dropbox / Yandex.Disk documentation and saw that responce includes may useful information (size, date, metadata, download link and so on..). All this information may be stored in String object and then parsed by means of an external helper class which will give us a JSON object. But, as you have told me, we can return a JSON object and thus omit external String -> JSON object conversion.